packages/core-types/src/Types.ts

```diff
@@ -20,17 +22,17 @@ export type RelayResponse = {
   connectionUrl?: string
 }
 export type UserOperation = {
-  sender: string
-  nonce: number
-  initCode: string
-  callData: string
-  callGasLimit: number
-  verificationGasLimit: number
-  preVerificationGas: number
-  maxFeePerGas: number
-  maxPriorityFeePerGas: number
-  paymasterAndData: string
-  signature: string
+  sender: string;
+  nonce: BigNumberish;
+  initCode: BytesLike;
+  callData: BytesLike;
+  callGasLimit: BigNumberish;
+  verificationGasLimit: BigNumberish;
+  preVerificationGas: BigNumberish;
```

Comment:
Why not use `BigNumber` here.

Reply:
these values are of type uint256 type so any thing should be acceptable Number, string, bignumber

Reply:
BigNumberish is good imo

Reply:
what's the advantage of using BigNumberish over BigNumber?

Reply:
flexibility, you can pass various types without manually creating a BigNumber.